### PR TITLE
fix: nextstep action greyed out

### PR DESCRIPTION
**Pull request by jianwei1**
# Description

### <samp>🤖 Generated by Copilot at 34d68db</samp>

This pull request updates the journey editor to handle the `NavigateAction` option more robustly. It also includes a minor change to `Step.tsx` that should be reverted.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/32387559/todos/6165464334)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [x] Go to a journey with more than one card
On the first card, add a button
The "Action" property for this button should have the "Next Step" option available

# Walkthrough

### <samp>🤖 Generated by Copilot at 34d68db</samp>

*  Add a function `noNextStepAvailable` to check if there is a next step for the current step in the journey editor ([link](https://github.com/JesusFilm/core/pull/1598/files?diff=unified&w=0#diff-5a35a228f99f223cb1f367e925c18fcfb801da1fc265692c8bda5c2fc742ea02R163-R175))
*  Use the function `noNextStepAvailable` to disable the `NavigateAction` option in the `Action` component ([link](https://github.com/JesusFilm/core/pull/1598/files?diff=unified&w=0#diff-5a35a228f99f223cb1f367e925c18fcfb801da1fc265692c8bda5c2fc742ea02L182-R195))
*  Remove the console log statement in the `Step` component ([link](https://github.com/JesusFilm/core/pull/1598/files?diff=unified&w=0#diff-52786729887fc8d8ea1f792f06dfe826f4cdb6ca0ce345493ae48379128ab2e6R30))

### CODE REVIEW

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Action/Action.tsx [hunk 1]
```diff
@@ -160,6 +160,19 @@ export function Action(): ReactElement {
     setAction(event.target.value)
   }
 
+  const noNextStepAvailable = (): boolean => {
+    const currentParentOrder = state.selectedStep?.parentOrder
+    const nextStep = state.steps?.find(
+      (step) =>
+        currentParentOrder != null &&
+        currentParentOrder + 1 === step.parentOrder
+    )
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Action/Action.tsx:169]: Heya, this looks amazing! Just thinking of further simplification. What would it look like to use directly? So meaning no function and no next step variables

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Action/Action.tsx [hunk 2]
```diff
@@ -160,6 +160,19 @@ export function Action(): ReactElement {
     setAction(event.target.value)
   }
 
+  const noNextStepAvailable = (): boolean => {
+    const currentParentOrder = state.selectedStep?.parentOrder
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Action/Action.tsx:164]: Are you able to destructure state?

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Action/Action.tsx [hunk 3]
```diff
@@ -179,8 +192,7 @@ export function Action(): ReactElement {
                   key={`button-action-${action.value}`}
                   value={action.value}
                   disabled={
-                    state.selectedStep?.nextBlockId == null &&
-                    action.value === 'NavigateAction'
+                    noNextStepAvailable() && action.value === 'NavigateAction'
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Action/Action.tsx:195]: Probs need to add or udpate tests for this

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Action/Action.tsx (hunk 1 above)
Comment (jianwei1) [reply apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Action/Action.tsx:169]: Have tried playing around with this and using it directly inside "disabled ={...}" on line 195, but it makes the statement too long. I think it's best if we keep it as a function

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Action/Action.tsx (hunk 3 above)
Comment (jianwei1) [reply apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Action/Action.tsx:195]: there is a test already('disables Next Step option if there is no next step'). I've updated the test data

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Action/Action.tsx [hunk 4]
```diff
@@ -160,6 +160,19 @@ export function Action(): ReactElement {
...
+    const currentParentOrder = state.selectedStep?.parentOrder
+    const nextStep = state.steps?.find(
+      (step) =>
+        currentParentOrder != null &&
+        currentParentOrder + 1 === step.parentOrder
+    )
+    if (nextStep != null) {
+      return false
+    }
+    return true
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Action/Action.tsx:173]:
```suggestion
    return nextStep == null
```

Comment (jianwei1) [reply apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Action/Action.tsx]: done!

**Review by GeronimoJohn — APPROVED** [review]

**Pull request merged**